### PR TITLE
Deploy with slack command

### PR DESCRIPTION
**Pull request by therealplato**
This PR parses "/broadway deploy foo bar" when sent through Slack and triggers the deployment.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -200,27 +202,61 @@ func (s *Server) postCommand(c *gin.Context) {
...
+	c.String(code, output)
 	return
 }
 
-func helperRunCommand(text string) (string, error) {
-	return "unimplemented :sadpanda:", nil
+// helperRunCommand takes the plaintext command, minus the leading /broadway
+// trigger, and returns statusCode, message, error for output to the user
+func helperRunCommand(s *Server, text string) (int, string, error) {
+	hint := `/broadway help: This message
```
Comment (bobbytables) [inline server/server.go]: This should be a constant.

File: server/server_test.go [hunk 2]
```diff
@@ -257,6 +257,23 @@ func TestPostCommandHelp(t *testing.T) {
...
+	if err := os.Setenv(slackTokenENV, testToken); err != nil {
+		t.Fatal(err)
+	}
+	w, server := helperSetupServer()
+	req, _ := http.NewRequest("POST", "/command", nil)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	form := url.Values{}
+	form.Set("token", testToken)
+	form.Set("command", "/broadway")
+	form.Set("text", "deploy foo")
```
Comment (bobbytables) [inline server/server_test.go]: Can't you do this when you initialize the `url.Values` struct?

Comment (therealplato) [inline server/server_test.go]: @macat and i switched to this syntax while troubleshooting form problems, I think we were having trouble setting the body properly from the url.Values struct

File: server/server.go [hunk 3]
```diff
@@ -200,27 +204,59 @@ func (s *Server) postCommand(c *gin.Context) {
...
-	c.String(http.StatusOK, output)
+	c.String(code, output)
 	return
 }
 
-func helperRunCommand(text string) (string, error) {
-	return "unimplemented :sadpanda:", nil
+// helperRunCommand takes the plaintext command, minus the leading /broadway
+// trigger, and returns statusCode, message, error for output to the user
+func helperRunCommand(s *Server, text string) (int, string, error) {
```
Comment (macat) [inline server/server.go]: simplify the name to maybe `runCommand`

**Comment by macat** [comment]
+1

File: server/server_test.go (hunk 2 above)
Comment (bobbytables) [inline server/server_test.go]: Interesting.

File: server/server.go [hunk 4]
```diff
@@ -200,28 +205,59 @@ func (s *Server) postCommand(c *gin.Context) {
...
+
+		_, err := doDeploy(s, commands[1], commands[2])
+		if err != nil {
+			return http.StatusInternalServerError, "Deployment failed", err
+		}
+		msg := fmt.Sprintf("Instance %s/%s deployed", commands[1], commands[2])
+		return http.StatusOK, msg, nil
+	default:
+		return http.StatusNotImplemented, "unimplemented :sadpanda:", nil
+	}
```
Comment (bobbytables) [inline server/server.go]: I think this ok for _for now_ but we really should be aiming for Open-Closed principle. Right now to add another slash command we have to modify this function, not ideal.

https://en.wikipedia.org/wiki/Open/closed_principle

**Comment by bobbytables** [comment]
:+1:

File: server/server.go (hunk 4 above)
Comment (therealplato) [inline server/server.go]: It's probably a good time to refactor it when we are adding CLI, otherwise we will be repeating ourselves

**Comment by bobbytables** [comment]
:+1:

**Pull request merged**